lib/java/desktop-environment/src/main/java/org/enso/desktopenvironment/WindowsTrashBin.java

```diff
@@ -0,0 +1,97 @@
+package org.enso.desktopenvironment;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import org.graalvm.nativeimage.StackValue;
+import org.graalvm.nativeimage.c.CContext;
+import org.graalvm.nativeimage.c.constant.CConstant;
+import org.graalvm.nativeimage.c.function.CFunction;
+import org.graalvm.nativeimage.c.struct.CField;
+import org.graalvm.nativeimage.c.struct.CStruct;
+import org.graalvm.nativeimage.c.type.CCharPointer;
+import org.graalvm.nativeimage.c.type.CTypeConversion;
+import org.graalvm.word.PointerBase;
+import org.slf4j.LoggerFactory;
+
+@CContext(WindowsTrashBin.ShellApi.class)
+final class WindowsTrashBin implements TrashBin {
+  @CConstant
+  public static native int FO_DELETE();
+
+  @CConstant
+  public static native int FOF_ALLOWUNDO();
+
+  @CConstant
+  public static native int FOF_NO_UI();
+
+  @Override
+  public boolean isSupported() {
+    return true;
+  }
+
+  @Override
+  public boolean moveToTrash(Path path) {
+    if (Platform.getOperatingSystem().isWindows())
```

Comment:
I'd say unnecessary, the same way as for macOS

Reply:
Same reason as for MacOS - native image build.